### DISCUSSION TITLE
Is there any method to get defined url path like a url_rule in flask

### DISCUSSION THREAD

**Discussion by warnus** [fastapi/fastapi#4527] [Questions]
Hello,
Thank you for your attention this question.

I wonder if it's possible that I get the defined url path like a flask url_rule.

For example, if I defined url like this in flask :
/info/<string:car_types>/<string:car_number>

I can get the url path using url_rule method.

Does someone knows the way in FastAPI?

Thank you.

**Comment by erdnaxeli** [comment]
To detail, this attribute `url_rule` is available on the request object with flask. Having it on the request object too would allow to use it in hooks (like for metrics).

**Reply by erdnaxeli** [reply]
I think it needs to be fixed on the starlette side, there is already an issue about it, it seems not that easy: https://github.com/encode/starlette/issues/685.

**Reply by warnus** [reply]
@erdnaxeli Thank you for the reply. I think so. If I want to get urLrule like flask thing, it need to change starlette code. Maybe, no way to implement url_rule right now. Thank you for the helping me.
